app/assets/javascripts/components/util/routes.jsx

```diff
@@ -20,32 +22,48 @@ const TrainingApp = lazy(() => import('../../training/components/training_app.js
 const ActiveCoursesHandler = lazy(() => import('../active_courses/active_courses_handler.jsx'));
 const CoursesByWikiHandler = lazy(() => import('../courses_by_wiki/courses_by_wiki_handler.jsx'));
 
-const routes = (
-  <Suspense fallback={<Loading />}>
-    <Routes>
-      <Route path="/onboarding/*" element={<Onboarding />} />
-      <Route path="/recent-activity/*" element={<RecentActivityHandler />} />
-      <Route path="/courses/:course_school/:course_title/*" element={<Course />} />
-      <Route path="/course_creator" element={<ConnectedCourseCreator />} />
-      <Route path="/users/:username" element={<UserProfile />} />
-      <Route path="/alerts_list" element={<AdminAlerts />} />
-      <Route path="/settings" element={<SettingsHandler />} />
-      <Route path="/article_finder" element={<ArticleFinder />} />
-      <Route path="/training/*" element={<TrainingApp />} />
-      <Route path="/tickets/dashboard" element={<TicketsHandler />} />
-      <Route path="/tickets/dashboard/:id" element={<TicketShowHandler />} />
-      <Route path="/campaigns/*" element={<CampaignsHandler />} />
-      <Route path="/tagged_courses/:tag/alerts" element={<TaggedCourseAlerts />} />
-      <Route index element={<DetailedCampaignList headerText={I18n.t('campaign.campaigns')} userOnly={true}/>} />
-      <Route path="/dashboard" element={<DetailedCampaignList headerText={I18n.t('campaign.campaigns')} userOnly={true}/>} />
-      <Route path="/explore" element={<Explore dashboardTitle={window.dashboardTitle}/>} />
-      <Route path="/active_courses" element={<ActiveCoursesHandler dashboardTitle={window.dashboardTitle}/>}/>
-      <Route path="/courses_by_wiki/:wiki_url" element={<CoursesByWikiHandler />}/>
+const routes = () => {
+  const location = useLocation();
+  const dispatch = useDispatch();
+  const refreshArgs = {
+    lastUserRequestTimestamp: useSelector(state => state.users.lastRequestTimestamp),
+    courseSlug: useSelector(state => state.course.slug),
+    limit: useSelector(state => state.articles.limit),
+    lastRequestArticleTimestamp: useSelector(state => state.articles.lastRequestTimestamp),
+    lastRequestAssignmentTimestamp: useSelector(state => state.assignments.lastRequestTimestamp)
+  };
 
-      {/* this prevents the "route not found" warning for pages which are server rendered */}
-      <Route path="*" element={<div style={{ display: 'none' }}/>} />
-    </Routes>
-  </Suspense>
-);
+  useEffect(() => {
```

Comment:
Whoops, I think the comment I meant to leave here got lost...

I think this should have some comments explaining what this useEffect is for and when the function itself will fire (separately from when `refreshData` will do something, which is based on timestamps).

Reply:
It'll fire when there is a change to the route - I've added a comment saying that. I'm not sure there's more to it than that :sweat_smile:

Reply:
Sorry, I should have been more clear. I think the comment should explain the strategy here: firing the effect whenever the route changes, but using logic within refreshData to determine whether to actually make the request... since this is a strategy we'll probably want to extend to other similar requests as well, by adding more to the refreshData logic.

This file is a pretty high level one in the component hierarchy, so being a little more explicit about what job it's doing will be particularly useful for newcomers to the codebase.